### PR TITLE
More small fixes

### PR DESCRIPTION
**Pull request by slavapestov**
rdar://problem/45962466, rdar://problem/23427259, rdar://problem/45962425

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

File: lib/Sema/TypeCheckGeneric.cpp [hunk 1]
```diff
@@ -33,19 +33,14 @@ using namespace swift;
...
-                           GenericSignatureBuilder *builder,
-                           GenericParamList *genericParams,
-                           GenericSignature *parentSig,
-                           TypeResolution resolution) {
+static void checkGenericParamList(TypeChecker &tc,
+                                  GenericSignatureBuilder *builder,
+                                  GenericParamList *genericParams,
+                                  GenericSignature *parentSig,
+                                  TypeResolution resolution) {
   // If there is a parent context, add the generic parameters and requirements
```
Comment (xedin) [inline lib/Sema/TypeCheckGeneric.cpp]: Maybe it would be appropriate to assert that builder is not a `nullptr` here just to be on the safe side?

Comment (slavapestov) [reply lib/Sema/TypeCheckGeneric.cpp]: I don’t think that’s necessary. We will hit a segfault immediately if it’s null.

**Pull request merged**